### PR TITLE
[Enhancement] support cast struct/map/array to json

### LINKED ISSUE #19320: Support type casting from struct to json
## Enhancement

Since 2.5 SR supported the `struct` type, without native storage.

The workaround should be, store it as JSON and cast struct to json when ingestion.

So the requirement is, support various type casting from struct to JSON:
- [x] Broker load/Stream load from parquet struct to JSON
- [ ] Support cast operator from struct to JSON

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19320

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/exprs/cast_expr.h [hunk 1]
```diff
@@ -149,4 +151,30 @@ class CastStructExpr final : public Expr {
...
 };
 
+/**
+ * Cast other type to string without float, double, string
+ */
+struct CastToString {
+    template <typename Type, typename ResultType>
+    static std::string apply(const Type& v) {
+        if constexpr (IsDate<Type> || IsTimestamp<Type> || IsDecimal<Type>) {
+            // DateValue, TimestampValue, DecimalV2
```
Comment (satanson) [inline be/src/exprs/cast_expr.h:161]: DecimalV2 is deprecated,   use decimal v3,  DecimalV3Cast::to_string to convert.

File: be/src/exprs/cast_expr_json.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            RETURN_IF_ERROR(cast_datum_to_json(elements, i, "", _builder));
+        }
+
+        if (!_builder->isClosed()) {
+            _builder->close();
+        }
+
+        return {};
+    }
+
```
Comment (satanson) [inline be/src/exprs/cast_expr_json.cpp:163]: do_visit for DecimalV3Column is absent

use DecimalNonDecimalCast::decimal_to

Comment (mofeiatwork) [reply be/src/exprs/cast_expr_json.cpp:163]: good suggestion

File: be/src/exprs/cast_expr.h (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exprs/cast_expr.h:161]: actually decimalV3 is supported but not in this code path, so don't need to worry about it

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/PrimitiveType.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/Type.java|2|2|100.00%|[]|

File: be/src/column/column_builder.h [hunk 3]
```diff
@@ -87,6 +87,14 @@ class ColumnBuilder {
         _column->append_default();
     }
 
+    void append_nulls(int count) {
+        _has_null = true;
+        for (int i = 0; i < count; i++) {
+            _null_column->append(DATUM_NULL);
```
Comment (silverbullet233) [inline be/src/column/column_builder.h:93]: can we use append_value_multiple_times instead of this loop?

**Review by satanson — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**